x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/utils/get_mode_display_text.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ADMIN, READ_ONLY, READ_WRITE, SEARCH, SEARCH_DISPLAY, WRITE_ONLY } from '../constants';
+import { IApiToken } from '../types';
+
+export const getModeDisplayText = (apiToken: IApiToken): string => {
+  const { read = false, write = false, type } = apiToken;
+
+  switch (type) {
+    case ADMIN:
+      return '--';
+    case SEARCH:
+      return SEARCH_DISPLAY;
+    default:
+      if (read && write) {
+        return READ_WRITE;
+      }
+      return write ? WRITE_ONLY : READ_ONLY;
```

Comment:
How do you feel.... about.... the dreaded nested ternary???

```js
return read && write ? READ_WRITE : write ? WRITE_ONLY : READ_ONLY;
```

Haha I'm actually super medium on it and I'm just fine with what you already have, just thought I'd be goofy and throw it out there. If you really like it or want a one-liner feel free to use it, if not just ignore me 😝

Reply:
Hahaha. It breaks my brain a little.